### DISCUSSION TITLE
Cannot POST, CANNOT GET

### DISCUSSION THREAD

**Discussion by kmostofi** [Mimetis/Dotmim.Sync#1175] [Q&A]
Hi,
After a year, I picked up the mobile app project. I followed the documentation and the sample. Create the API, published in Azure, and create a test project to see the result. However, I keep getting this Exception for this line of code:

`var s1 = await agent.SynchronizeAsync(Dotmim.Sync.Enumerations.SyncType.Normal, progress);`

`at Dotmim.Sync.SyncAgent.SynchronizeAsync(String scopeName, SyncSetup setup, SyncType syncType, SyncParameters parameters, CancellationToken cancellationToken, IProgress`1 progress)
   at Dotmim.Sync.SyncAgent.SynchronizeAsync(String scopeName, SyncSetup setup, SyncType syncType, SyncParameters parameters, CancellationToken cancellationToken, IProgress`1 progress)
   at Program.SynchronizeAsync()`

What am I missing?
Please advice.

Kind regards,
Kamal

**Comment by Mimetis** [comment]
Sorry, not enough info to help you here.
Do you have more details %?

**Reply by kmostofi** [reply]
Thanks for getting back to me. This is the code in the startup.cs 
`public class Startup(IConfiguration configuration)
{
    private IConfiguration Configuration { get; } = configuration;

    public void ConfigureServices(IServiceCollection services)
    {
        services.AddControllers();

        services.AddDistributedMemoryCache();
        services.AddSession(options => options.IdleTimeout = TimeSpan.FromMinutes(30));

        var connectionString = Configuration.GetSection("ConnectionStrings")["DefaultConnection"];

        var options = new SyncOptions
        {
            SnapshotsDirectory = "C:\\Tmp\\Snapshots",
            BatchSize = 2000,
        };

        var tables = new string[] {
            "printoptions",
            "printorder"};

        services.AddSyncServer<SqlSyncProvider>(connectionString, tables, options);
    }

    public void Configure(IApplicationBuilder app, IWebHostEnvironment env)
    {
        if (env.IsDevelopment())
            app.UseDeveloperExceptionPage();

        app.UseHttpsRedirection();
        app.UseRouting();
        app.UseAuthorization();
        app.UseSession();
        app.UseEndpoints(endpoints => endpoints.MapControllers());
    }
}`

and this is method from my test project:

`private static async Task SynchronizeAsync()
      {
          var serverOrchestrator = new WebRemoteOrchestrator("AzureUrl");
          var clientProvider = new SqliteSyncProvider(clientConnectionString);

          var options = new SyncOptions
          {
              BatchSize = 1000
          };

          var agent = new SyncAgent(clientProvider, serverOrchestrator, options);

          do
          {
              try
              {
                  var progress = new SynchronousProgress<ProgressArgs>(args => Console.WriteLine($"{args.ProgressPercentage:p}:\t{args.Message}"));

                  var s1 = await agent.SynchronizeAsync(Dotmim.Sync.Enumerations.SyncType.Normal, progress);

                  Console.WriteLine(s1);
}
              catch (Exception ex)
              {
                  Console.WriteLine(ex.Message);
              }

          } while (Console.ReadKey().Key != ConsoleKey.Escape);

          Console.WriteLine("End");
      }`

I hope these information helps.

**Reply by Mimetis** [reply]
not really
Your code is pretty standard, and this code works when I test it
I need a way to reproduce your error.

The only thing I see is `new WebRemoteOrchestrator("AzureUrl");`

I guess you are replacing `AzureUrl` with a correct connection string.

Are you able to query your **Azure SQL instance** from a console application using a standard `SqlConnection` and `SqlCommand` ?

**Reply by kmostofi** [reply]
That is right, I replace it with the azure uri. hmmm, did I confuse myself and should use the SQL connectionstring? I am using https://name.azurewebsites.net/

Yes, I am able to run a query using Azure SQL instance from a console using a SqlConnection and a SqlCommand.

**Reply by Mimetis** [reply]
So before using a Web based approach, are you able to make a sync using a direct sync provider ?

``` cs
var serverOrchestrator = new SqlSyncProvider(azureConnectionString);
var clientProvider = new SqliteSyncProvider(clientConnectionString);
var agent = new SyncAgent(clientProvider, serverOrchestrator, options);
```

**Reply by kmostofi** [reply]
I will try it and will confirm the result.

**Reply by kmostofi** [reply]
Yes, I was able to sync it directly. However, when I looked at the data, I noticed Version columns are Null for all records. This is supposed to have a byte[] data, which recently came to my attention. Previously I thought the Version data type was a string!

Any thoughts on why the Version come through as Null? 

I was using Microsoft DataSync library and always threw exceptions for Version data. Now, it is more clear that it is null.

**Reply by kmostofi** [reply]
Hi,
I republished the web API. The GET method responds 200 but the POST method throws 404 Not Found. I checked the setting and also went through the samples in the github.  

Am I missing any settings?

**Reply by Mimetis** [reply]
I dont know
Maybe share a github repository where you are reproducing the error will help

**Reply by kmostofi** [reply]
I added to a repository that contains 2 projects. 
Thanks.
